tools/Get-AppInsightsResources.ps1

```diff
@@ -0,0 +1,21 @@
+[CmdletBinding()]
+param(
+    [string]$AppInsightsNamePrefix
+)
+
+$AppInsightsResources = Get-AzResource -ResourceType "Microsoft.Insights/components" -Name "$AppInsightsNamePrefix*"
+
+$AppInsightsResourcesArray = @()
+
+foreach ($AppInsightsResource in $AppInsightsResources) {
+    $AppInsightsResourceObject = New-Object PSObject
+    $AppInsightsResourceObject | Add-Member -NotePropertyMembers @{
+        appInsightsName             = $AppInsightsResource.Name
+        appInsightsResourceGroup    = $AppInsightsResource.ResourceGroupName
+    }
+    $AppInsightsResourcesArray += $AppInsightsResourceObject
+}
+
+$AppInsightsResourcesArrayString = $AppInsightsResourcesArray | ConvertTo-Json -Compress
+
+Write-Output ("##vso[task.setvariable variable=APP_INSIGHTS_RESOURCES_ARRAY_STRING]$($AppInsightsResourcesArrayString)")
```

Comment:
Why are the outer most set of braces needed?

Reply:
Good spot, I've removed them